hack/testing/logging.sh

```diff
@@ -0,0 +1,239 @@
+
+install_registry
+wait_for_registry
+
+######### logging specific code starts here ####################
+
+### create and deploy the logging component pods ###
+masterurlhack=",MASTER_URL=https://172.30.0.1:443"
+OS_O_A_L_DIR=${OS_O_A_L_DIR:-$OS_ROOT/test/extended/origin-aggregated-logging}
+os::cmd::expect_success "oc new-project logging"
+os::cmd::expect_success "oc secrets new logging-deployer nothing=/dev/null"
+os::cmd::expect_success "echo 'apiVersion: v1
+kind: ServiceAccount
+metadata:
+  name: logging-deployer
+secrets:
+- name: logging-deployer
+' | oc create -f -"
+os::cmd::expect_success "oc policy add-role-to-user edit system:serviceaccount:logging:logging-deployer"
+if [ -n "$USE_LOGGING_DEPLOYER" ] ; then
+    imageprefix="docker.io/openshift/origin-"
+elif [ -n "$USE_LOGGING_DEPLOYER_SCRIPT" ] ; then
+    pushd $OS_O_A_L_DIR/deployment
+    IMAGE_PREFIX="openshift/origin-" PROJECT=logging ./run.sh
+    popd
+    imageprefix=
+else
+    os::cmd::expect_success "oc process \
+       -f $OS_O_A_L_DIR/hack/templates/dev-builds.yaml \
+       -v LOGGING_FORK_URL=$GIT_URL,LOGGING_FORK_BRANCH=$GIT_BRANCH \
+       | oc create -f -"
+    os::cmd::expect_success "wait_for_builds_complete"
+    imageprefix=`oc get is | awk '$1 == "logging-deployment" {print gensub(/^([^/]*\/logging\/).*$/, "\\\1", 1, $2)}'`
+    sleep 5
+fi
+log_priv_user=$(create_valid_file add-privileged-logging-user.yml)
+os::cmd::expect_success "oc get scc/privileged -o yaml > $log_priv_user"
+os::cmd::expect_success "echo '- system:serviceaccount:logging:aggregated-logging-fluentd' >> $log_priv_user"
+os::cmd::expect_success "oc replace -f $log_priv_user"
+sleep 5
+os::cmd::expect_success "oadm policy add-cluster-role-to-user cluster-reader \
+                      system:serviceaccount:logging:aggregated-logging-fluentd"
+sleep 5
+if [ ! -n "$USE_LOGGING_DEPLOYER_SCRIPT" ] ; then
+    os::cmd::expect_success "oc process \
+                          -f $OS_O_A_L_DIR/deployment/deployer.yaml \
+                          -v IMAGE_PREFIX=$imageprefix,KIBANA_HOSTNAME=kibana.example.com,ES_CLUSTER_SIZE=1,PUBLIC_MASTER_URL=https://localhost:8443${masterurlhack} \
+                          | oc create -f -"
+    os::cmd::try_until_text "oc get pods logging-deployment-1-build --template='{{.status.phase}}'" "Succeeded"
+    os::cmd::try_until_text "oc get pods -l component=deployer" "Completed" "$(( 3 * TIME_MIN ))"
+fi
+# see if expected pods are running
+os::cmd::try_until_text "oc get pods -l component=es" "Running" "$(( 3 * TIME_MIN ))"
+os::cmd::try_until_text "oc get pods -l component=kibana" "Running" "$(( 3 * TIME_MIN ))"
+os::cmd::try_until_text "oc get pods -l component=curator" "Running" "$(( 3 * TIME_MIN ))"
+# this fails because the imagestreams already exist
+os::cmd::expect_failure_and_text "oc process logging-support-template | oc create -f -" "already exists"
+# scale up a fluentd pod
+os::cmd::try_until_success "oc get dc logging-fluentd"
```

Comment:
likewise regarding #63 we'll be using a daemonset, so we'll make sure the daemonset exists and then we'll want to `oc label node/<node_name> logging-infra=fluentd` to deploy instead of scaling up